services/svc_simpleurl.py

```diff
@@ -0,0 +1,60 @@
+# This file is part of photoframe (https://github.com/mrworf/photoframe).
+#
+# photoframe is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# photoframe is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with photoframe.  If not, see <http://www.gnu.org/licenses/>.
+#
+from base import BaseService
+import os
+import requests
+
+class SimpleUrl(BaseService):
+  SERVICE_NAME = 'Simple URL'
+  SERVICE_ID   = 3
+
+
+  def __init__(self, configDir, id, name):
+    BaseService.__init__(self, configDir, id, name, needConfig=False, needOAuth=False)
+
+
+  def helpKeywords(self):
+    return 'Each item is a URL that should return a single image. The URL may contain the terms "{width}" and/or "{height}" which will be replaced by numbers describing the size of the display.'
+
+
+  def fetchImage(self, destinationFile, supportedMimeTypes, displaySize):
+    urlList = list(self.getKeywords())
+    if len(urlList) == 0:
+      return {'mimetype' : None, 'error' : 'No URLs have been specified', 'source': None}
+
+    Url = urlList[self.getRandomKeywordIndex()]
+
+    Url = Url.replace('{width}', str(displaySize['width']))
+    Url = Url.replace('{height}', str(displaySize['height']))
+
+    r = requests.get(Url)
```

Comment:
You should use `self.requestUrl(Url, destination=destinationFile)` instead, it's simpler and removes the need for importing requests here. This function returns a dict where one of the keys is `status` which holds the error code.

Granted, that function isn't documented today so I understand why you didn't use it, but there ya go 😄

Reply:
I did start there but then lifted the code into here because `self.requestUrl()` doesn't return a MIME type and I didn't want to assume the URL was returning a jpeg. I considered fixing that but python is new to me and I wasn't sure that I understood the potential consequences of doing that. Now that I've got this far I think I can see that I could trivially add the MIME type to the dict without risk to the rest of the code so I could make that change.

What do you think?

Reply:
You make a great point, I was relying on the JSON provided by google. So I've added it to the master branch now. You get the convenience of a `mimetype` key but also full access to all headers in the `headers` key, which should cover most cases I hope.

Reply:
Excellent. Thank you :)
I've further simplified my service to make use of that.